### PR TITLE
Fix failing test //tensorflow/python:function_test

### PR DESCRIPTION
**Pull request by yongtang**
@benoitsteiner It seems `//tensorflow/python:function_test` is failing with my last commit from #13998 (sorry about that).

It looks like the `function_test` created a graph with python through a helper class and invoked tests through c api.

I am not familiar with the `function_test`, though as `ClipByValue` is actually hidden, and is exposed through python only, with the gradient defined in python as well, I think replacing it will fix the issue?

Please take a look. Again, really sorry for the caused inconvenience.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by yongtang** [comment]
Looks like the `//tensorflow/tools/api/tests:api_compatibility_test` is also failing, caused by
```
ERROR:tensorflow:2 differences found between API and golden.
ERROR:tensorflow:Issue 1        : New object tensorflow.nn.rnn_cell.SRUCell found (added).
ERROR:tensorflow:Issue 2        : Change detected in python object: tensorflow.nn.rnn_cell.
```

The issue is from another PR #13978 with introduced the SRU implementation.

It seems that `tensorflow/core/api_def/update_api_def.sh` only adds the ops in c++ but does not update in golden.

I run with
```
bazel-bin/tensorflow/tools/api/tests/api_compatibility_test \
          --update_goldens True
```

and the issue seems to be fixed.

Pushed though to the PR, though not sure this is the right fix as I am not familiar with the API changes.

**Comment by tjingrant** [comment]
@yongtang thanks, I was not aware of the script for API update. Sorry for your trouble.

**Comment by caisq** [comment]
@martinwicke @ebrevdo, FYI: PR https://github.com/tensorflow/tensorflow/pull/13978 involved API changes, but was merged without an "API review" tag. This led to API compatibility test failures that @yongtang fixed in this PR. Please see whether it is okay to approve this PR or revert https://github.com/tensorflow/tensorflow/pull/13978 first and roll it forward later after the proper API change review process.

File: tensorflow/tools/api/golden/tensorflow.nn.rnn_cell.-s-r-u-cell.pbtxt [hunk 1]
```diff
@@ -0,0 +1,152 @@
+path: "tensorflow.nn.rnn_cell.SRUCell"
```
Comment (gunan) [inline tensorflow/tools/api/golden/tensorflow.nn.rnn_cell.-s-r-u-cell.pbtxt]: Please revert this change. It should have gone through its own API review. I will revert the culprit change to be submitted with an API review.

**Comment by yongtang** [comment]
A PR #15455 has been created to re-address `clip_by_value`. I will close this PR. Thanks all for the help!

**Pull request closed without merging**